### ISSUE TITLE
model.tune(): How is population size defined and how can it be controlled?

### ISSUE THREAD

**Issue by Aditya-365** [ultralytics/ultralytics#24279]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

## Description
I am using Ultralytics YOLO's built-in hyperparameter tuning via `model.tune()` (without Ray Tune), and I would like to better understand how the genetic algorithm is configured.

## Question
How is the **population size** defined in the default genetic algorithm?

- Is it fixed internally?
- Is it derived from the number of iterations?
- Is there a way to explicitly control or override it?

## What I tried
I ran:
~~~python
from ultralytics import YOLO

model = YOLO("yolo12n.pt")
model.tune(data="data.yaml", epochs=50, iterations=40, use_ray=False)
~~~

## Environment
OS: Linux
Python: 3.11.15
Ultralytics: 8.3.63
Torch: 2.7.0+cu128
GPU: NVIDIA RTX 5090

### Additional

The tuning process runs successfully, but I want to better understand how the search space is explored and whether adjusting population size could improve results.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Aditya-365, thank you for your interest in Ultralytics 🚀! This is an automated response to help get things moving quickly, and an Ultralytics engineer will also assist you soon.
We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.
Since this is a tuning/configuration ❓ question about `model.tune()`, please share as much detail as possible, including:
- Your full tuning command or script
- Any relevant console logs
- Whether you are using default search settings or custom hyperparameters
- What behavior led you to investigate population size
If this becomes a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.
If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).
Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.
## Upgrade
Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:
```bash
pip install -U ultralytics
```
## Environments
YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/get-started/locally/) preinstalled):
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>
## Status
<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>
If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by glenn-jocher** [comment]
It’s not using a user-settable population in the usual GA sense: the built-in tuner runs `iterations` sequential trials, and each new trial is generated from up to the top `n=9` prior results in [`Tuner._mutate()`](https://docs.ultralytics.com/reference/engine/tuner/#ultralytics.engine.tuner.Tuner._mutate) plus BLX-α crossover in [`Tuner._crossover()`](https://docs.ultralytics.com/reference/engine/tuner/#ultralytics.engine.tuner.Tuner._crossover), so `iterations` controls total evaluations, not population size, and there’s currently no public `model.tune()` argument to override that without editing `ultralytics/engine/tuner.py`.
